packages/aws-amplify-react-native/src/InAppMessaging/components/hooks/useMessageImage/useMessageImage.ts

```diff
@@ -44,24 +46,24 @@ export default function useMessageImage(image: InAppMessageImage, layout: InAppM
 	useEffect(() => {
 		if (hasImage) {
 			prefetchNetworkImage(src).then((loadingState) => {
-				// get image size once loaded
 				if (loadingState === 'loaded') {
+					// get image size once loaded
 					Image.getSize(
 						src,
 						(width, height) => {
-							setImageDimensions(getImageDimensions(height, width, layout));
+							setImageDimensions(getLayoutImageDimensions(height, width, layout));
 						},
 						(error) => {
 							logger.error(`Unable to retrieve size for image: ${error}`);
 
 							// set dimension values to 0 if size retrieval failure
-							setImageDimensions({ height: 0, width: 0 });
+							setImageDimensions(FAILURE_IMAGE_DIMENSIONS);
 						}
 					);
+				} else {
+					// set dimension values to 0 if prefetch failure
```

Comment:
Super nit: If for some weird reason failure dimensions has reason to be something else one day
```suggestion
					// set to failure dimension values if prefetch fails
```